backend/core/socketio_config.py

```diff
@@ -21,10 +36,135 @@
     socketio_server=sio_server, socketio_path=SOCKETIO_PATH
 )
 
+cookie = SimpleCookie()
+
+
+def get_value_or_raise_error(data_dict, key, error_msg=None):
+    try:
+        value = data_dict[key]
+    except KeyError:
+        if error_msg is None:
+            error_msg = f"Key '{key}' not found in message"
+        raise KeyError(error_msg)
+    return value
+
+
+def check_conversation_exist_and_generate_queue_message(
+    session: Session, msg: dict, user_phone_number: str
+):
+    conversation_envelope = get_value_or_raise_error(
+        msg, "conversation_envelope"
+    )
+    client_phone_number = get_value_or_raise_error(
+        conversation_envelope, "client_phone_number"
+    )
+
+    conversation_exist = session.exec(
+        select(Chat_Session)
+        .join(User)
+        .join(Client)
+        .where(
+            User.phone_number == user_phone_number,
+            Client.phone_number == client_phone_number,
+        )
+    ).first()
+
+    if conversation_exist:
+        iso_timestamp = datetime.now(timezone.utc).isoformat()
+        sender_role = get_value_or_raise_error(
+            conversation_envelope, "sender_role"
+        )
+        platform = get_value_or_raise_error(conversation_envelope, "platform")
+
+        queue_message = queue_message_util.create_queue_message(
+            message_id=get_value_or_raise_error(
+                conversation_envelope, "message_id"
+            ),
+            client_phone_number=client_phone_number,
+            user_phone_number=user_phone_number,
+            sender_role=(
+                Sender_Role_Enum[sender_role.upper()]
+                if sender_role
+                else sender_role
+            ),
+            sender_role_enum=Sender_Role_Enum,
+            platform=(
+                Platform_Enum[platform.upper()] if platform else platform
+            ),
+            platform_enum=Platform_Enum,
+            body=get_value_or_raise_error(msg, "body"),
+            media=get_value_or_raise_error(msg, "media"),
+            context=get_value_or_raise_error(msg, "context"),
+            transformation_log=get_value_or_raise_error(
+                msg, "transformation_log"
+            ),
+            timestamp=iso_timestamp,
+        )
+        return queue_message
+    return conversation_exist
+
+
+def handle_incoming_message(session: Session, message: dict):
```

Comment:
Hi @kjkoster , is this part of code solve the [Matching extension worker and farmer](https://app.asana.com/0/1207412452922189/1207784298476921/f) task?

Reply:
Yes, looking good.